pkg/cache/v3/linear.go

```diff
@@ -470,3 +464,41 @@ func (cache *LinearCache) NumDeltaWatches() int {
 	defer cache.mu.Unlock()
 	return len(cache.deltaWatches)
 }
+
+// TriggerDeltaWatchesForNode forces go-control-plane to re-send some resources to a particular envoy node.
+// This is useful to workaround an envoy bug https://github.com/envoyproxy/envoy/issues/13009
+// Currently, envoy expects control plane to re-sent all EDS data after a CDS update.
+// In SOTW we have some other ways to do that, because envoyat least send a new ServiceDiscoveryRequest after CDS update.
+// In delta, however, control plane in this case isn't notified in any way.
+// This method can be used in the following way:
+// 1. Use calbacks to detect when CDS update happens.
+// 2. Parse DeltaDiscoveryResponse payload and figure out what ClusterLoadAssignment resources must be updated.
+// 3. Call TriggerDeltaWatchesForNode for the affected node and pass the list of  ClusterLoadAssignment names.
+func (cache *LinearCache) TriggerDeltaWatch(streamID int64, resourceNames []string) {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+	w, found := cache.deltaWatches[streamID]
+	if !found {
+		return
+	}
+	resourceNamesContains := func(name string) bool {
+		for _, r := range resourceNames {
+			if r == name {
+				return true
+			}
+		}
+		return false
+	}
+	versions := map[string]string{}
+	// We want to force go-control-plane to re-send some resources to envoy.
+	// We do that by providing a state where corresponding resources are marked as 'stale'
+	for k, v := range w.StreamState.GetResourceVersions() {
+		if resourceNamesContains(k) {
+			versions[k] = ""
+		} else {
+			versions[k] = v
+		}
+	}
+	cache.respondDelta(w.Request, w.Response, w.StreamState)
```

Comment:
Don't you need to cancel the watch here? This will use the response buffer that is expected to never be taken in the current model. If you call this from the same typeUrl (so the same cache), the response channel is already being processed and will no longer be watched in https://github.com/s-matyukevich/go-control-plane/blob/main/pkg/server/delta/v3/server.go#L177